### PR TITLE
Transform circle fence to polygon fence for display

### PR DESCRIPTION
**Pull request by mikems66**

### CODE REVIEW

File: modern/src/map/GeofenceMap.js [hunk 1]
```diff
@@ -4,6 +4,15 @@ import { useEffect, useState } from 'react';
 import { map } from './Map';
 import { useEffectAsync } from '../reactHelper';
 import { reverseCoordinates } from './mapUtil';
+import circle from '@turf/circle';
+
+
+const circleToPolygon = (item) => {
+  let coord = item.area.replace(/CIRCLE|\(|\)|,/g, " ").trim().split(/ +/);
```
Comment (tananaev) [inline modern/src/map/GeofenceMap.js:11]:
```suggestion
  let coordinates = item.area.replace(/CIRCLE|\(|\)|,/g, " ").trim().split(/ +/);
```

File: modern/src/map/GeofenceMap.js [hunk 2]
```diff
@@ -4,6 +4,15 @@ import { useEffect, useState } from 'react';
...
+  var options = {steps: 64, units: 'meters'};
```
Comment (tananaev) [inline modern/src/map/GeofenceMap.js:12]:
```suggestion
  var options = { steps: 64, units: 'meters' };
```

I would also reduce the number of steps.

File: modern/src/map/GeofenceMap.js [hunk 3]
```diff
@@ -4,6 +4,15 @@ import { useEffect, useState } from 'react';
...
+  let c = circle([Number(coord[1]), Number(coord[0])], Number(coord[2]), options);
```
Comment (tananaev) [inline modern/src/map/GeofenceMap.js:13]: Please use full words for variable names. Maybe `polygon` or something here.

File: modern/src/map/GeofenceMap.js [hunk 4]
```diff
@@ -72,9 +81,10 @@ const GeofenceMap = () => {
   }, []);
 
   useEffect(() => {
+    let circleFence = geofences.filter((item => item.area.indexOf('CIRCLE') > -1)).map(circleToPolygon);
     map.getSource(id).setData({
       type: 'FeatureCollection',
-      features: geofences.map(item => [item.name, reverseCoordinates(wellknown(item.area))]).filter(([, geometry]) => !!geometry).map(([name, geometry]) => ({
```
Comment (tananaev) [inline modern/src/map/GeofenceMap.js:77]: Should we implement this circle logic in `wellknown` instead? Where is that defined?

File: modern/src/map/GeofenceMap.js (hunk 1 above)
Comment (mikems66) [reply modern/src/map/GeofenceMap.js]: ok

File: modern/src/map/GeofenceMap.js (hunk 2 above)
Comment (mikems66) [reply modern/src/map/GeofenceMap.js]: ok, steps is now 32

File: modern/src/map/GeofenceMap.js (hunk 3 above)
Comment (mikems66) [reply modern/src/map/GeofenceMap.js]: ok

File: modern/src/map/GeofenceMap.js (hunk 4 above)
Comment (mikems66) [reply modern/src/map/GeofenceMap.js:77]: "wellknown" is a Node module that creates geojson object from text, and as I understand it, it doesn't handle circles, so I don't think we can use it.

Comment (tananaev) [reply modern/src/map/GeofenceMap.js:77]: In that case we should probably move the whole logic outside of this UI component.

Comment (mikems66) [reply modern/src/map/GeofenceMap.js:77]: Is this ok; 
new function createGeofenceFeatures in map/mapUtil.js, which will create features for all geometry types.

Comment (tananaev) [reply modern/src/map/GeofenceMap.js:77]: I think you forgot to push the changes.

Comment (mikems66) [reply modern/src/map/GeofenceMap.js:77]: Here it is, actually new function name is geofenceToFeature.

**Comment by tananaev** [comment]
Looks good. Merged. Thank you for the contribution.

**Pull request merged**